### PR TITLE
Remove internal api usages from describer

### PR DESCRIPTION
**Pull request by soltysh**
Replaces #22993, builds on top of #23018 so ignore first commit.
/cc @damemi

### CODE REVIEW

**Comment by soltysh** [comment]
OK, this builds on top of https://github.com/openshift/origin/pull/23077 and https://github.com/openshift/origin/pull/23079 so only last 4 commits matter. I'm working on the remaining unit tests for describer.

**Comment by soltysh** [comment]
This builds on top of https://github.com/openshift/origin/pull/23079, so only last 4 commits matter.

File: hack/import-restrictions.json [hunk 1]
```diff
@@ -455,17 +455,7 @@
       "vendor/k8s.io/kubernetes/pkg/util",
       "vendor/github.com/davecgh/go-spew/spew",
 
-      "github.com/openshift/origin/pkg/authorization/generated",
-      "github.com/openshift/origin/pkg/image/generated",
       "github.com/openshift/origin/pkg/image/util",
-      "github.com/openshift/origin/pkg/oauth/generated",
-      "github.com/openshift/origin/pkg/project/generated",
```
Comment (deads2k) [inline hack/import-restrictions.json]: so happy!

**Comment by deads2k** [comment]
/hold

test-cmd failure is legit

**Comment by soltysh** [comment]
After digging more it turned out being simpler than I thought :)

**Comment by soltysh** [comment]
Failure in test-cmd are not related to this PR.
/retest

**Comment by soltysh** [comment]
/hold
it looks like the `oc process ruby-helloworld-sample -o describe` is legit, testing...

**Pull request merged**